test/test_web.py

```diff
@@ -57,22 +58,48 @@ class AbstractWebTest(unittest.TestCase):
     driver: webdriver.Firefox
 
     def setUp(self) -> None:
+        if not self.share_selenium_web_driver():
+            opts = selenium.webdriver.firefox.options.Options()
+            opts.add_argument("-headless")
+            self.driver = webdriver.Firefox(options=opts)
+
         self.server_runner = InterfaceThreadRunner(shc.web.WebServer, "localhost", 42080, 'index')
         self.server = self.server_runner.interface
 
     @classmethod
     def setUpClass(cls) -> None:
-        opts = selenium.webdriver.firefox.options.Options()
-        opts.add_argument("-headless")
-        cls.driver = webdriver.Firefox(options=opts)
+        if cls.share_selenium_web_driver():
+            opts = selenium.webdriver.firefox.options.Options()
+            opts.add_argument("-headless")
+            cls.driver = webdriver.Firefox(options=opts)
 
     def tearDown(self) -> None:
+        if not self.share_selenium_web_driver():
+            self.driver.close()
+            self.driver.quit()
+
         self.server_runner.stop()
 
     @classmethod
     def tearDownClass(cls) -> None:
-        cls.driver.close()
-        cls.driver.quit()
+        if cls.share_selenium_web_driver():
+            cls.driver.close()
+            cls.driver.quit()
+
+    @staticmethod
+    def share_selenium_web_driver() -> bool:
+        """Should the selenium driver be shared or not.
```

Comment:
Maybe rather: "Determine whether the selenium driver shall be shared between all consecutive tests in this class"